### PR TITLE
feat: new end to end tutorial 

### PR DESCRIPTION
**Pull request by marekq**

### CODE REVIEW

File: tutorials/edi-to-json/handler.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  console.log(putBucketData.$metadata.httpStatusCode);
+  return;
+}
+
+// GET the bucket object
+async function getBucketObject(bucketName: string, bucketObjectKey: string) {
+
+  var bodyOutput;
+  var statusCode;
+  var statusError;
```
Comment (simalexan) [inline tutorials/edi-to-json/handler.ts:59]: maybe not use vars, but const or let?

File: tutorials/edi-to-json/handler.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+// Main function handler
+export async function handler(event, _context) {
+    
+  // Get the bucket object key from event input
+  const bucketName = event.Records[0].s3.bucket.name || "default";
+  const bucketObjectKey = event.Records[0].s3.object.key || "test.txt";
+
+  // If the S3 object key does not start with "input/", return an error
+  if (!bucketObjectKey.startsWith("input/")) {
+    const msg = "Bucket name " + bucketObjectKey + " does not start with input/";
```
Comment (simalexan) [inline tutorials/edi-to-json/handler.ts:100]: why not use back ticks?

File: tutorials/edi-to-json/handler.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    console.log(msg);
+    return msg;
+
+  } else {
+    console.log("Bucket name " + bucketName + " starts with input, proceeding");
```
Comment (simalexan) [inline tutorials/edi-to-json/handler.ts:105]: same here?

File: tutorials/edi-to-json/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,213 @@
+#!/bin/bash
```
Comment (pcholakov) [inline tutorials/edi-to-json/deploy.sh:1]: Now that we have a JS-friendly SDK, I'd love to see these deployment commands as npm run scripts. I think it's a lot cleaner and customers can reuse the code more easily for their own needs. WDYT?

Comment (marekq) [reply tutorials/edi-to-json/deploy.sh:1]: This is definitely something I want to do, but didn't get to yet. 

I have a different proposal; the best end solution is Stedi CLI. We can already do part of the workflow with it with the current version, but still need to expand CreateFunction and BucketNotification. 

If I were to use Stedi, I'd probably use CLI as a starting user for my first functions. For more advanced use cases or automation, you would use the SDK. But in either case, SDK or CLI should replace this temporary bash script - hopefully very soon.

Comment (marekq) [reply tutorials/edi-to-json/deploy.sh:1]: Regarding your second comment - we could ship this for day zero users and actively work on CLI _and_ SDK. It's well understood there are potential API changes coming and that this solution is very minimum effective dose.

File: tutorials/edi-to-json/handler.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    output_format: "jedi@2.0"
+  }
+
+  // Create a new Axios request
+  const request = await axios.request({
+    method: "POST",
+    url: "https://edi-core.stedi.com/2021-06-05/translate",
+    headers: {
+      "Content-Type": "application/json",
+      "Authorization": "Key " + process.env.STEDI_API_KEY
```
Comment (pcholakov) [inline tutorials/edi-to-json/handler.ts:25]: There's this and a few more instances where you could use template strings if you wanted to be consistent.

**Comment by marekq** [comment]
I like that suggestion a lot Pavel - let me work on it and include it.

**Comment by pcholakov** [comment]
Sorry, I thought I had approved this but apparently not – giving it a 🚢 to unblock.

**Review by pcholakov — APPROVED** [review]

**Pull request merged**